Library/Homebrew/cmd/update.sh

```diff
@@ -331,6 +345,13 @@ EOS
     UPSTREAM_BRANCH="$(upstream_branch)"
     # the refspec ensures that the default upstream branch gets updated
     (
+      if [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
+      then
+        # Skip taps without formulae.
+        FORMULAE="$(find "$DIR" -maxdepth 1 -name '*.rb' -or -name 'Formula' -print -quit)"
+        [ -z "$FORMULAE" ] && exit
```

Comment:
All the previous variables in here aren't local so although a local variable might be nice I personally prefer consistency.